core/src/test/java/org/apache/iceberg/rest/requests/TestCommitTransactionRequestParser.java

```diff
@@ -0,0 +1,245 @@
+
+  @Test
+  public void invalidTableIdentifier() {
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"ns1.table1\" : \"ns1.table1\"}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing field: identifier");
+
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\" : {}}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing string: name");
+
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\" : { \"name\": 23}}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse to a string value: name: 23");
+  }
+
+  @Test
+  public void invalidRequirements() {
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\":{\"namespace\":[\"ns1\"],\"name\":\"table1\"}}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing field: requirements");
+
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\":{\"namespace\":[\"ns1\"],\"name\":\"table1\"},"
+                        + "\"requirements\":[23],\"updates\":[]}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse update requirement from non-object value: 23");
+
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\":{\"namespace\":[\"ns1\"],\"name\":\"table1\"},"
+                        + "\"requirements\":[{}],\"updates\":[]}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse update requirement. Missing field: type");
+
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\":{\"namespace\":[\"ns1\"],\"name\":\"table1\"},"
+                        + "\"requirements\":[{\"type\":\"assert-table-uuid\"}],\"updates\":[]}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing string: uuid");
+  }
+
+  @Test
+  public void invalidMetadataUpdates() {
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\":{\"namespace\":[\"ns1\"],\"name\":\"table1\"},\"requirements\":[]}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing field: updates");
```

Comment:
I think that this is currently allowed and I think we should continue to allow it. I have two reasons:

1. It's easy for a service to detect a no-op request and return a 200 code immediately. That's much better than having the service fail because a client wasn't smart enough to detect that the request wasn't necessary.
2. There will be requests with only requirements produced by multi-table commits. For example, if you read a table then you have to validate that it has the same or an equivalent table state. That will produce a `AssertRefSnapshotID` requirement with no changes.

Reply:
makes sense, I've updated the code so that updates/requirements are both optional in the parser